tools/tests/add_and_update.py

```diff
@@ -0,0 +1,129 @@
+#! /usr/bin/env python3
+
+import sys
+
+from util import *
+from init_tests import init_project
+
+project_name="aou_project"
+project_path=os.path.join("./",project_name)
+
+def expect_subvalue(ll, name, tag, value):
+   """Inside ll, find the list that contains `name` and test the value stored in that list for the given tag against value.
+
+   :param ll  An array of lists.
+   :param name  The name to search for.
+   :tag  The tag that contains the value we to test.
+   :value  The expected value.
+   """
+
+   for i in ll:
+      if i["name"] == name:
+         return i[tag] == value
+   return False
+
+
+def setup_project():
+   """Setup the project ensuring there is a fresh project file.
+   """
+
+   shutil.rmtree(project_path, ignore_errors=True)
+   init_project(project_name, "v1.0.0", project_path)
+   assert(os.path.isdir(project_path))
```

Comment:
I have run pylint and got something that might improve readability in python (for all `assert`):
add_and_update.py:32:0: C0325: Unnecessary parens after 'assert' keyword (superfluous-parens)
add_and_update.py:42:0: C0325: Unnecessary parens after 'assert' keyword (superfluous-parens)
add_and_update.py:47:0: C0325: Unnecessary parens after 'assert' keyword (superfluous-parens)
add_and_update.py:57:0: C0325: Unnecessary parens after 'assert' keyword (superfluous-parens)
add_and_update.py:58:0: C0325: Unnecessary parens after 'assert' keyword (superfluous-parens)
add_and_update.py:63:0: C0325: Unnecessary parens after 'assert' keyword (superfluous-parens)
add_and_update.py:64:0: C0325: Unnecessary parens after 'assert' keyword (superfluous-parens)
add_and_update.py:74:0: C0325: Unnecessary parens after 'assert' keyword (superfluous-parens)
add_and_update.py:75:0: C0325: Unnecessary parens after 'assert' keyword (superfluous-parens)
add_and_update.py:101:0: C0325: Unnecessary parens after 'assert' keyword (superfluous-parens)
add_and_update.py:102:0: C0325: Unnecessary parens after 'assert' keyword (superfluous-parens)
```suggestion
   assert os.path.isdir(project_path)
```